### PR TITLE
Update GitHub workflows

### PR DESCRIPTION
**Pull request by flavio**
Use the new reusable workflows and GitHub Actions provided by the
Kubewarden project.

Also, improve the Makefile to have better targets.

### CODE REVIEW

**Comment by flavio** [comment]
Converted to be a draft, cargo-generate is not working as expected...

**Comment by flavio** [comment]
Everything is working now, this is ready to be reviewed

File: Makefile [hunk 1]
```diff
@@ -10,10 +15,15 @@ fmt:
 lint:
 	cargo clippy -- -D warnings
 
+.PHONY: e2e-tests
+e2e-tests: annotated-policy.wasm
+	bats e2e.bats
```
Comment (jvanz) [inline Makefile:20]: Should not we add a fake test file to avoid errors right after the users create their projects?

Comment (jvanz) [inline Makefile:20]: Should not we add a fake test file to avoid errors right after the users create their projects?

Comment (flavio) [reply Makefile:20]: This is correct! Let me add something

**Comment by flavio** [comment]
This can be reviewed again.

Changes since the last time are inside of 2 dedicated commits:
  * ensure all the files are rendered correctly at generation time. Some files had the template placeholders not processed at all
  * provide a simple e2e test that works out of the box

**Review by jvanz — APPROVED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  test:
+    name: run tests and linters
+    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-rust.yml@v1
+
+  release:
+    needs: test
+    permissions:
+      # Required to push to GHCR
+      contents: write
```
Comment (viccuad) [inline .github/workflows/release.yml:21]: the current comment is somewhat incorrect, as `contents` is required for creating GH Releases. I would prefer something like this:
```
      # Required to create GH releases
      contents: write
      # Required to push to GHCR
      packages: write
      # Required by cosign keyless signing
      id-token: write
```

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,6 @@
 /target
+policy.wasm
```
Comment (viccuad) [inline .gitignore:2]: Could be simplified to `*.wasm`

**Review by viccuad — APPROVED** [review]
LGTM, with nitpicks

File: .github/workflows/release.yml (hunk 2 above)
Comment (flavio) [reply .github/workflows/release.yml:21]: fixed

File: .gitignore (hunk 3 above)
Comment (flavio) [reply .gitignore]: done

**Comment by flavio** [comment]
ready to be reviewed again

**Review by jvanz — APPROVED** [review]

**Review by viccuad — APPROVED** [review]

**Pull request merged**